tests/performance/profiling.py

```diff
@@ -0,0 +1,77 @@
+import timeit
+import tracemalloc
+from multiprocessing import Process, Value
+
+import pandas as pd
+
+
+def _profile_time(method, dataset, iterations=100):
+    total_time = 0
+    for _ in range(iterations):
+        start_time = timeit.default_timer()
+        method(dataset)
+        total_time += timeit.default_timer() - start_time
+
+    return total_time / iterations
+
+
+def _set_memory_for_method(method, dataset, peak_memory):
+    tracemalloc.start()
+    method(dataset)
+    peak_memory.value = tracemalloc.get_traced_memory()[1]
+    tracemalloc.stop()
+
+
+def _profile_memory(method, dataset):
+    peak_memory = Value('i', 0)
+    profiling_process = Process(target=_set_memory_for_method, args=(method, dataset, peak_memory))
+    profiling_process.start()
+    profiling_process.join()
+    return peak_memory.value
+
+
+def profile_transformer(transformer, dataset_generator, transform_size, fit_size=None):
+    """
+    Function to profile a transformer.
+
+    This function will get the total time and peak memory
+    for the ``fit``, ``transform`` and ``reverse_transform``
+    methods of the provided transformer against the provided
+    dataset.
+
+    Args:
+        transformer (Transformer):
+            Transformer instance.
+        dataset_generator (DatsetGenerator):
+            DatasetGenerator instance.
+        transform_size (int):
+            Number of rows to generate for ``transform`` and ``reverse_transform``.
+        fit_size (int or None):
+            Number of rows to generate for ``fit``. If None, use ``transform_size``.
+    """
+    transformer_instance = transformer()
```

Comment:
I think this line should not be required, since `transformed` is expected to already be an instance of the class.